corehq/apps/locations/models.py

```diff
@@ -678,15 +678,16 @@ def save(self, *args, **kwargs):
             ]
             self.site_code = generate_code(self.name, all_codes)
 
+        # Set the UUID here so we can save to SQL first (easier to rollback)
+        if not self.location_id:
+            self._id = self.get_db().server.next_uuid()
```

Comment:
just thinking out loud here: is it ever possible to already have an _id but not a location_id? if yes, then we could end up changing the _id of the doc and creating another one accidentally. i wonder if we should check for `if not self._id`?